File: app/models/organization.py
import uuid
from sqlalchemy import String, DateTime
from sqlalchemy.orm import Mapped, mapped_column
from datetime import datetime

from app.models.base import Base


class Organization(Base):
    __tablename__ = "organizations"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    name: Mapped[str] = mapped_column(String(160), nullable=False)
    slug: Mapped[str] = mapped_column(String(160), unique=True, nullable=False)

    country_code: Mapped[str | None] = mapped_column(String(2))
    timezone: Mapped[str] = mapped_column(String(64), default="UTC")

    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)

File: noa-frontend/src/types/organization.ts
export type Organization = {
  id: string;
  name: string;
  slug: string;
  country_code: string | null;
  timezone: string;
  created_at: string;
};

export type OrganizationCreatePayload = {
  name: string;
  slug: string;
  country_code: string | null;
  timezone: string;
};
